src/workspace.rs

```diff
@@ -151,77 +224,11 @@ pub fn apply_edit(
-                            ResourceOp::Create(op) => {
-                                let path = op.uri.to_file_path().unwrap();
-                                let ignore_if_exists = if let Some(options) = op.options {
-                                    !options.overwrite.unwrap_or(false)
-                                        && options.ignore_if_exists.unwrap_or(false)
-                                } else {
-                                    false
-                                };
-                                if !(ignore_if_exists && path.exists())
-                                    && fs::write(&path, []).is_err()
-                                {
-                                    error!(
-                                        "Failed to create file: {}",
-                                        path.to_str().unwrap_or("")
-                                    );
-                                    applied = true;
-                                }
-                            }
-                            ResourceOp::Delete(op) => {
-                                let path = op.uri.to_file_path().unwrap();
-                                if path.is_dir() {
-                                    let recursive = if let Some(options) = op.options {
-                                        options.recursive.unwrap_or(false)
-                                    } else {
-                                        false
-                                    };
-                                    if recursive {
-                                        if fs::remove_dir_all(&path).is_err() {
-                                            error!(
-                                                "Failed to delete directory: {}",
-                                                path.to_str().unwrap_or("")
-                                            );
-                                            applied = true;
-                                        }
-                                    } else if fs::remove_dir(&path).is_err() {
-                                        error!(
-                                            "Failed to delete directory: {}",
-                                            path.to_str().unwrap_or("")
-                                        );
-                                        applied = true;
-                                    }
-                                } else if path.is_file() && fs::remove_file(&path).is_err() {
-                                    error!(
-                                        "Failed to delete file: {}",
-                                        path.to_str().unwrap_or("")
-                                    );
-                                    applied = true;
-                                }
+                        DocumentChangeOperation::Op(op) => {
+                            if apply_document_resource_op(&meta, op, ctx) {
```

Comment:
Should it be `applied = apply_document_resource_op(&meta, op, ctx)`? Because default value for `applied` is true (line 213) and if `apply_document_resource_op` would return `false`, `applied` will stay `true` in the current implementation.

Reply:
From what I could tell, `applied` does nothing right now and is just there for future use? I don't think it actually does anything yet.

Reply:
We send it back to the language server:

```
pub fn apply_edit_from_server(id: Id, params: Params, ctx: &mut Context) {
    let params: ApplyWorkspaceEditParams = params.parse().expect("Failed to parse params");
    let meta = ctx.meta_for_session();
    let response = apply_edit(meta, params.edit, ctx);
    ctx.reply(id, Ok(serde_json::to_value(response).unwrap()));
}
```

Even if langserv does nothing with it we probably should provide correct value.

Reply:
Sorry, what I mean is that in the current code, it's never set to false. It's only set to true at the start and true in every assignment.

Reply:
Ah, now I get it. I think it's a long-running bug then, it's probably meant to be `false` by default.

Reply:
Okay, there is a problem with this `applied` but it's not resolved by setting it to the `false` by default. Let's solve it later.